core/new-gui/src/app/workspace/service/execute-workflow/execute-workflow.service.ts

```diff
@@ -119,6 +119,13 @@ export class ExecuteWorkflowService {
           errorMessages[entry[0]] = entry[1];
         });
         return { state: ExecutionState.Failed, errorMessages: errorMessages };
+      // TODO: Merge WorkflowErrorEvent and ErrorEvent
+      case 'ErrorEvent':
+        const backendErrorMessages: Record<string, string> = {};
+        Object.entries(event.errorMap).forEach(entry => {
+          backendErrorMessages[entry[0]] = entry[1];
```

Comment:
some more examples are appreciated, for example, when the code is not written inside the controller (like in a separate class, or in principal, or in worker), especially when the error is in the worker, where the worker is not on the same machine as the controller.

Reply:
Yes. I am planning extending the functionality to worker and other classes in the next PR. I need to think about how to do that.